### PR TITLE
Add auto approve for Service Prinicipals in TF 

### PR DESCRIPTION
**Pull request by goldblatt**
We need to auto approve `terraform apply` when using a service principal and not use that if it's a user input

### CODE REVIEW

**Comment by goldblatt** [comment]
info here: https://learn.hashicorp.com/tutorials/terraform/automate-terraform

File: cloud/shared/bin/lib/terraform.sh [hunk 1]
```diff
@@ -18,14 +19,25 @@ function terraform::perform_apply() {
...
       -backend-config="${BACKEND_VARS_FILENAME}"
   fi
   
-  terraform \
-    -chdir="${TERRAFORM_TEMPLATE_DIR}" \
-    apply \
-    -var-file="${TF_VAR_FILENAME}"
+  if azure::is_service_principal; then
+    terraform \
+        -chdir="${TERRAFORM_TEMPLATE_DIR}" \
```
Comment (kmcnellis) [inline cloud/shared/bin/lib/terraform.sh:29]: [nit] Might be worth putting this into a variable, so we know to edit both commands together (and just add the extra flag on the end)

**Review by kmcnellis — APPROVED** [review]

**Comment by kmcnellis** [comment]
Not sure this PR name is accurate?

File: cloud/shared/bin/lib/terraform.sh [hunk 2]
```diff
@@ -7,6 +7,7 @@
 #   TERRAFORM_TEMPLATE_DIR
 #   BACKEND_VARS_FILENAME
 #   TF_VAR_FILENAME
+#
```
Comment (bion) [inline cloud/shared/bin/lib/terraform.sh:10]: nit: 
```suggestion
```

**Review by bion — APPROVED** [review]

**Review by bion — DISMISSED** [review]

File: cloud/shared/bin/lib/terraform.sh [hunk 3]
```diff
@@ -10,22 +18,23 @@
 #######################################
 function terraform::perform_apply() {
   if [[ "${CIVIFORM_MODE}" == "dev" ]]; then
-    terraform -chdir="${TERRAFORM_TEMPLATE_DIR}" init -upgrade
+    ${TERRAFORM_BASE_COMMAND} init -upgrade
```
Comment (shanemc-goog) [inline cloud/shared/bin/lib/terraform.sh:21]: Need quotes around all the new ${} commands similar to the below command you're keeping.

**Review by shanemc-goog — CHANGES_REQUESTED** [review]

File: cloud/shared/bin/lib/terraform.sh (hunk 3 above)
Comment (goldblatt) [reply cloud/shared/bin/lib/terraform.sh:21]: ahh i thought it would make a string

Comment (goldblatt) [reply cloud/shared/bin/lib/terraform.sh:21]: ya this doesn't work as expected-- i can switch it

Comment (goldblatt) [reply cloud/shared/bin/lib/terraform.sh]: okay i think this is now what the style guide reccomends

**Review by bion — APPROVED** [review]

File: cloud/shared/bin/lib/terraform.sh (hunk 3 above)
Comment (shanemc-goog) [reply cloud/shared/bin/lib/terraform.sh]: #sorry.....  So many characters

**Review by shanemc-goog — APPROVED** [review]

**Pull request merged**